pkg/bosh/converter/job_factory.go

```diff
@@ -150,40 +144,44 @@ func (f *JobFactory) VariableInterpolationJob() (*ejv1.ExtendedJob, error) {
 }
 
 // InstanceGroupManifestJob generates the job to create an instance group manifest
-func (f *JobFactory) InstanceGroupManifestJob() (*ejv1.ExtendedJob, error) {
-	containers := make([]corev1.Container, len(f.Manifest.InstanceGroups))
+func (f *ConcreteJobFactory) InstanceGroupManifestJob(manifest bdm.Manifest) (*ejv1.ExtendedJob, error) {
+	containers := make([]corev1.Container, len(manifest.InstanceGroups))
+
+	// ExtendedJob will always pick the latest version for versioned secrets
+	desiredManifestName := names.DesiredManifestName(manifest.Name, "1")
```

Comment:
In other parts of the code base I see a `names.DesiredManifestName(manifest.Name, "")`, so sometimes we want a specific version, sometimes we just give some version ("1") and when we give "" we just want the manifest name.

Reply:
If you use `names.DesiredManifestName(manifest.Name, "")`, it's mean that you get secret prefix and then use `versionedSecretStore.Latest` to fetch the latest desired manifest.

In this case, it's **another** path to set resources' references that contains versioned desired manifest like placeholder, and let `GetReconciles` func knows this object has versioned secret and determine current versioned secret is referenced by this object. 

Details info is in `reference` package: `pkg/kube/util/reference/reconciles.go`